update github actions and readme

zkamvar opened the pull request.

zkamvar committed:
update github actions and readme

zkamvar committed:
add epidict for needs/check

zkamvar committed:
add epidict to needs

zkamvar committed:
add ggplot2 to imports

zkamvar committed:
fix tidyselect error

zkamvar committed:
update zcurve to require ggplot2

zkamvar committed:
move ggplot2 to imports since we import stuff

zkamvar commented:
This is absolutely bizzarre. The tests seem to want to revert to the version before this PR, but if I run it again, it's okay 😨 

``` r
devtools::test("~/Documents/Git/R4EPIs/epikit")
#> ℹ Testing epikit
#> ✔ | F W S  OK | Context
#> 
#> ⠏ |         0 | add_weights_cluster                                             
#> ⠋ |   1     0 | add_weights_cluster                                             
#> ⠸ |   2     2 | add_weights_cluster                                             
#> ⠦ |   2     5 | add_weights_cluster                                             
#> ✔ |   2     5 | add_weights_cluster [0.4s]
#> ────────────────────────────────────────────────────────────────────────────────
#> Warning ('test-add_weights_cluster.R:40'): add_weights will ignore clusters and households by default
#> Using an external vector in selections was deprecated in tidyselect 1.1.0.
#> ℹ Please use `all_of()` or `any_of()` instead.
#>   # Was:
#>   data %>% select(clus_id_x)
#> 
#>   # Now:
#>   data %>% select(all_of(clus_id_x))
#> 
#> See <https://tidyselect.r-lib.org/reference/faq-external-vector.html>.
#> Backtrace:
#>   1. epikit::add_weights_cluster(...)
#>        at test-add_weights_cluster.R:40:2
#>   3. tidyr:::unite.data.frame(...)
#>   4. tidyselect::eval_select(expr(c(...)), data)
#>   5. tidyselect:::eval_select_impl(...)
#>   9. tidyselect:::vars_select_eval(...)
#>  10. tidyselect:::walk_data_tree(expr, data_mask, context_mask)
#>  11. tidyselect:::eval_c(expr, data_mask, context_mask)
#>  12. tidyselect:::reduce_sels(node, data_mask, context_mask, init = init)
#>  13. tidyselect:::walk_data_tree(new, data_mask, context_mask)
#>  14. tidyselect:::eval_sym(expr, data_mask, context_mask)
#> 
#> Warning ('test-add_weights_cluster.R:40'): add_weights will ignore clusters and households by default
#> Using an external vector in selections was deprecated in tidyselect 1.1.0.
#> ℹ Please use `all_of()` or `any_of()` instead.
#>   # Was:
#>   data %>% select(hh_id_x)
#> 
#>   # Now:
#>   data %>% select(all_of(hh_id_x))
#> 
#> See <https://tidyselect.r-lib.org/reference/faq-external-vector.html>.
#> Backtrace:
#>   1. epikit::add_weights_cluster(...)
#>        at test-add_weights_cluster.R:40:2
#>   3. tidyr:::unite.data.frame(...)
#>   4. tidyselect::eval_select(expr(c(...)), data)
#>   5. tidyselect:::eval_select_impl(...)
#>   9. tidyselect:::vars_select_eval(...)
#>  10. tidyselect:::walk_data_tree(expr, data_mask, context_mask)
#>  11. tidyselect:::eval_c(expr, data_mask, context_mask)
#>  12. tidyselect:::reduce_sels(node, data_mask, context_mask, init = init)
#>  13. tidyselect:::walk_data_tree(new, data_mask, context_mask)
#>  14. tidyselect:::eval_sym(expr, data_mask, context_mask)
#> ────────────────────────────────────────────────────────────────────────────────
#> 
#> ⠏ |         0 | age-categories                                                  
#> ⠏ |         0 | age_categories()                                                
#> ✔ |        12 | age_categories()
#> 
#> ⠏ |         0 | group_age_categories                                            
#> ✔ |        10 | group_age_categories
#> 
#> ⠏ |         0 | find_breaks                                                     
#> ✔ |        12 | find_breaks
#> 
#> ⠏ |         0 | gen_polygon                                                     
#> ✔ |         1 | gen_polygon
#> 
#> ⠏ |         0 | helpers                                                         
#> ✔ |         4 | helpers
#> 
#> ⠏ |         0 | inline_fun                                                      
#> ⠼ |        25 | inline_fun                                                      
#> ✔ |        28 | inline_fun [0.1s]
#> 
#> ⠏ |         0 | proportion                                                      
#> ⠏ |         0 | proportion() tests                                              
#> ⠏ |        30 | proportion() tests                                              
#> ⠴ |        36 | proportion() tests                                              
#> ✔ |        37 | proportion() tests [0.3s]
#> 
#> ⠏ |         0 | relabel_proportions                                             
#> ✔ |         3 | relabel_proportions
#> 
#> ⠏ |         0 | unite_ci                                                        
#> ✔ |         7 | unite_ci
#> 
#> ⠏ |         0 | zcurve                                                          
#> ⠏ |         0 | zcurve tests                                                    
#> ✔ |   1     3 | zcurve tests
#> ────────────────────────────────────────────────────────────────────────────────
#> Warning ('test-zcurve.R:5'): (code run outside of `test_that()`)
#> Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
#> ℹ Please use `linewidth` instead.
#> Backtrace:
#>  1. epikit::zcurve(dat, zee)
#>       at test-zcurve.R:5:0
#>  2. ggplot2::stat_function(...)
#>       at epikit/R/zcurve.R:41:2
#>  3. ggplot2::layer(...)
#>  4. ggplot2:::deprecate_soft0(...)
#> ────────────────────────────────────────────────────────────────────────────────
#> 
#> ══ Results ═════════════════════════════════════════════════════════════════════
#> Duration: 1.3 s
#> 
#> [ FAIL 0 | WARN 3 | SKIP 0 | PASS 122 ]
devtools::test("~/Documents/Git/R4EPIs/epikit")
#> ℹ Testing epikit
#> ✔ | F W S  OK | Context
#> 
#> ⠏ |         0 | add_weights_cluster                                             
#> ⠙ |         2 | add_weights_cluster                                             
#> ⠼ |         5 | add_weights_cluster                                             
#> ✔ |         5 | add_weights_cluster [0.2s]
#> 
#> ⠏ |         0 | age-categories                                                  
#> ⠏ |         0 | age_categories()                                                
#> ✔ |        12 | age_categories()
#> 
#> ⠏ |         0 | group_age_categories                                            
#> ✔ |        10 | group_age_categories
#> 
#> ⠏ |         0 | find_breaks                                                     
#> ✔ |        12 | find_breaks
#> 
#> ⠏ |         0 | gen_polygon                                                     
#> ✔ |         1 | gen_polygon
#> 
#> ⠏ |         0 | helpers                                                         
#> ✔ |         4 | helpers
#> 
#> ⠏ |         0 | inline_fun                                                      
#> ⠦ |        27 | inline_fun                                                      
#> ✔ |        28 | inline_fun [0.1s]
#> 
#> ⠏ |         0 | proportion                                                      
#> ⠏ |         0 | proportion() tests                                              
#> ⠋ |        31 | proportion() tests                                              
#> ⠦ |        37 | proportion() tests                                              
#> ✔ |        37 | proportion() tests [0.2s]
#> 
#> ⠏ |         0 | relabel_proportions                                             
#> ✔ |         3 | relabel_proportions
#> 
#> ⠏ |         0 | unite_ci                                                        
#> ✔ |         7 | unite_ci
#> 
#> ⠏ |         0 | zcurve                                                          
#> ⠏ |         0 | zcurve tests                                                    
#> ✔ |         3 | zcurve tests
#> 
#> ══ Results ═════════════════════════════════════════════════════════════════════
#> Duration: 0.9 s
#> 
#> [ FAIL 0 | WARN 0 | SKIP 0 | PASS 122 ]
```

<sup>Created on 2023-01-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

aspina7 commented:
hey @zkamvar - sorry, was in a busy period, anything I can do to help with this?

zkamvar commented:
I'm not sure. This is in a state I've never seen before.

aspina7 commented:
Hmmmm is possible I made a booboo when moving over functions from {sitrep} to here

maelle commented on R/zcurve.R:
```diff
@@ -39,7 +39,7 @@ zcurve <- function(x, zscore) {
+    stat_density(aes(x = !! rlang::enquo(zscore), color = "Observed"), linewidth = 1,
                  geom = "line") +
     stat_function(fun     = stats::dnorm,
                   args    = list(mean = 0, sd = 1), size = 1,
```
@zkamvar this still uses size, is that the issue?

maelle replied:
and then it'd make sense, depending on how you run the tests, that you don't get the warning every time.

https://github.com/tidyverse/ggplot2/blob/15cb2fbc1de1c906fe5264d3cef85c2a9338f375/R/utilities.r#L717

> warnings are only issued once every 8 hours to avoid overwhelming the user.

https://lifecycle.r-lib.org/reference/deprecate_soft.html

zkamvar replied:
(edit, Zhian has forgotten that his nose is right in front of him) ~But the thing is, this PR fixes that issue. You can see in the diff `size` changes to `linewidth` 🤔~

zkamvar replied:
OMG I SEE IT NOW 😨

zkamvar commented on R/zcurve.R:
```diff
@@ -39,7 +39,7 @@ zcurve <- function(x, zscore) {
+    stat_density(aes(x = !! rlang::enquo(zscore), color = "Observed"), linewidth = 1,
                  geom = "line") +
     stat_function(fun     = stats::dnorm,
                   args    = list(mean = 0, sd = 1), size = 1,
```
```suggestion
                  args    = list(mean = 0, sd = 1), linewidth = 1,
```

zkamvar committed:
update linewidth arg

Thank you to @maelle for pointing out the solution that's been staring at me the whole time

zkamvar committed:
fix one problem and now a new problem :weary:

zkamvar commented:
Now that I've fixed one tidyselect problem, I've got a new one that we cannot use them with `tidyr::unite()` and it is _frustrating_ :fire: :fire: :fire:

aspina7 commented:
> Now that I've fixed one tidyselect problem, I've got a new one that we cannot use them with `tidyr::unite()` and it is _frustrating_ 🔥 🔥 🔥

no real need for `tidyr::unite()` here - we could just use {glue} no? Is dependency already anyway...

zkamvar committed:
replace unite with paste0(a, "_", b)

At some point, `tidyr::unite()` stopped working with selected variables
and was throwing the error "Can't rename variables in this context"

Because we are only joining two variables, I've decided to just manually
do this and throw out the damn tidyselect evaluation as this does not
affect user input

zkamvar committed:
make sure external data is present

zkamvar commented:
There is still a problem because one of our examples uses `sitrep::gen_population()`, which probably should live here. @aspina7 can you address this in this PR?

aspina7 committed:
swap {sitrep} out of example

aspina7 commented:
turns out I had already moved the `gen_*()` functions here - just didnt update the example sorry!

aspina7 committed:
updating documentation

aspina7 commented:
going to merge for now - as checks all passed and want to reorganise, moving the [proportion ](https://github.com/R4EPI/epikit/blob/master/R/cfr.R) (ar, cfr, and cmr) functions to {epitabulate}
